editor/editor_themes.cpp

```diff
@@ -218,8 +218,15 @@ void editor_register_and_generate_icons(Ref<Theme> p_theme, bool p_dark_theme =
 	// Generate icons.
 	if (!p_only_thumbs) {
 		for (int i = 0; i < editor_icons_count; i++) {
+			float icon_scale = EDSCALE;
+
+			// Always keep the DefaultProjectIcon at the default size
+			if (strcmp(editor_icons_names[i], "DefaultProjectIcon") == 0) {
```

Comment:
If we write 
```suggestion
			if (String(editor_icons_names[i]) == "DefaultProjectIcon")) {
```

Reply:
That would solve the "aesthetic" problem, but now we're allocating, copying and freeing a string on each iteration which feels unnecessary when we simply want to compare a few string constants.

Reply:
Also we can change type of `editor_icons_names`. I change array type to `String` in editor_icons.gen.h file.

Reply:
Yes, but that file is automatically generated so that would have to change as well, but that still doesn't change the fact that you're now allocating heap memory for all the strings in the `editor_icon_names` array even though that's avoidable.
I don't see why using the `String` class is necessary, while it does let us use the more familiar `==` for string comparison, the pay-off is not worth it in my opinion.

Maybe @Calinou can give us his opinion on the matter.

Reply:
So far, I've always used `String()` to perform string equality comparisons. It seems we don't use `strcmp()` in Godot's codebase often.

Have you benchmarked the difference in both debug and release builds? See the [Microbenchmarking snippet](https://docs.godotengine.org/en/latest/development/cpp/common_engine_methods_and_macros.html#microbenchmarking).

Reply:
I'll try benchmarking it

Reply:
Alright I have hyperfined the different configurations
`strcmp` is ~200 milliseconds faster in debug and ~60 milliseconds faster in release.
![ConEmu64_2020-09-22_15-22-08](https://user-images.githubusercontent.com/13891859/93887790-6fb50280-fce7-11ea-93f0-4733b59d3573.png)

Reply:
Alright, let's keep `strcmp()` then :slightly_smiling_face: